src/slipcover/slipcover.py

```diff
@@ -231,13 +235,18 @@ def both(f, field):
 class Slipcover:
     def __init__(self, immediate: bool = False,
                  d_miss_threshold: int = 50, branch: bool = False,
-                 disassemble: bool = False, source: Optional[List[str]] = None):
+                 disassemble: bool = False, source: List[str] | None = None, exclude_lines: Set[str] | None = None):
```

Comment:
Using the vertical bar here breaks support for Python 3.8 and 3.9; please use `Optional[]` instead.